pmlc/target/x86/stencil_tpp_binary.cc

```diff
@@ -50,6 +50,27 @@ Optional<TppOperand> getTppOperand(TOp op, Block *block,
 
   return TppOperand{op.getMemRef(), outerValueMap.getAffineMap(), tileMap};
 }
+
+bool shapeCompatible(MemRefType output, MemRefType input1, MemRefType input2) {
+  if (!output.hasStaticShape() || !input1.hasStaticShape() ||
+      !input2.hasStaticShape())
+    return false;
+  if (output.getRank() != input1.getRank() ||
+      output.getRank() != input2.getRank())
+    return false;
+
+  ArrayRef<int64_t> outShape = output.getShape();
+  ArrayRef<int64_t> inputShape1 = input1.getShape();
+  ArrayRef<int64_t> inputShape2 = input2.getShape();
+
+  for (int64_t i = 0; i < outShape.size(); i++) {
```

Comment:
ArrayRef has a != operator, so you can just say outShape != inputShape1.  This also automatically checks the length, so it would make the .size() comparisons above redundant as well.

Reply:
Ok sounds good, I can make this change, thanks